### PR TITLE
Add benchmark reproducing large static scrolling content

### PR DESCRIPTION
**Pull request by yjbanov**
## Description

Add benchmark that reproduces https://github.com/flutter/flutter/issues/42987, i.e. a large piece of static content (a `Picture` that never changes) is scrolled. As the clip slides along the picture it causes repaints. Since our DOM operations are not culled, it causes jank proportional to the size of the picture.

I will send a separate fix in flutter/engine, but want to get the benchmark first so we can validate the fix later.

## Related Issues

https://github.com/flutter/flutter/issues/42987

### CODE REVIEW

**Review by mdebbar — APPROVED** [review]
LGTM

**Review by ferhatb — APPROVED** [review]
Other than comment /// The number of rows should be such that , lgtm

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_dynamic_clip_on_static_picture.dart [hunk 1]
```diff
@@ -0,0 +1,109 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'recorder.dart';
+import 'test_data.dart';
+
+/// The number of rows should be such that
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply dev/benchmarks/macrobenchmarks/lib/src/web/bench_dynamic_clip_on_static_picture.dart]: Oops. Leftover. I moved this check into the constructor. Replaced with dartdocs.

File: dev/benchmarks/macrobenchmarks/lib/src/web/bench_dynamic_clip_on_static_picture.dart [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+///   picture with a static clip.
+/// * https://github.com/flutter/flutter/issues/42987, which this benchmark is
+///   based on.
+class BenchDynamicClipOnStaticPicture extends SceneBuilderRecorder {
+  BenchDynamicClipOnStaticPicture() : super(name: benchmarkName) {
+    // If the scrollable extent is too small, the benchmark may end up
+    // scrolling the picture out of the clip area entirely, resulting in
+    // bogus metric vaules.
+    const double maxScrollExtent = kMaxSampleCount * kScrollDelta;
+    const double pictureHeight = kRows * kRowHeight;
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply dev/benchmarks/macrobenchmarks/lib/src/web/bench_dynamic_clip_on_static_picture.dart]: They are here for validation only, and they don't supply any new configuration for this benchmark. The ones at the top can be manually tweaked, unlike these that will always remain derivatives of the top-level constants.

**Pull request merged**